### PR TITLE
Properly handle whitespace for Infinity and NaN

### LINKED ISSUE #244: Properly handle whitespace for Infinity and NaN

### PR DESCRIPTION
**Pull request by victorporof**
Fixes #244.

Signed-off-by: Victor Porof <vporof@mozilla.com>

### CODE REVIEW

File: edn/src/edn.rustpeg [hunk 1]
```diff
@@ -32,10 +32,10 @@ pub nil -> Value =
     "nil" { Value::Nil }
 
 pub nan -> Value =
-    "#f" __ "NaN" { Value::Float(OrderedFloat(NAN)) }
+    "#f " __ "NaN" { Value::Float(OrderedFloat(NAN)) }
 
 pub infinity -> Value =
-    "#f" __ s:$(sign) "Infinity" {
+    "#f " __ s:$(sign) "Infinity" {
```
Comment (rnewman) [inline edn/src/edn.rustpeg]: These are technically incorrect: EDN allows `','` as whitespace, so `#f,1.0` is valid (and you should add a test for this case). You want:

```
"#f" whitespace+ s:$(sign)
```

for this rule, and

```
"#f" whitespace+ "NaN"
```

for `nan`. That will also prevent users from putting a comment between a tag and its referent, which I think is a good thing!

File: edn/src/types.rs [hunk 2]
```diff
@@ -56,7 +56,6 @@ impl Display for Value {
             Integer(v) => write!(f, "{}", v),
             BigInteger(ref v) => write!(f, "{}N", v),
             Float(ref v) => {
-                // TODO: make sure float syntax is correct.
```
Comment (rnewman) [inline edn/src/types.rs]: Did you check that `format`'s rules for floats match EDN's? E.g., exponent capitalization and sign?

File: edn/src/types.rs [hunk 3]
```diff
@@ -56,7 +56,6 @@ impl Display for Value {
...
                 if *v == OrderedFloat(f64::INFINITY) {
                     write!(f, "#f {}", "+Infinity")
```
Comment (rnewman) [inline edn/src/types.rs]: You can just do

```
write!(f, "#f +Infinity")
```

for these – no need to use a format expression at all.

**Comment by victorporof** [comment]
Addressed comments.

**Comment by victorporof** [comment]
@rnewman or @ncalexan gentle review ping.

**Pull request merged**